FirebaseAppDistribution/Sources/FIRAppDistribution.m

```diff
@@ -129,65 +134,69 @@ - (void)signInTesterWithCompletion:(void (^)(NSError *_Nullable error))completio
 }
 
 - (void)signOutTester {
-  [FIRAppDistributionAuthPersistence clearAuthState];
+  NSError *error;
+  [FIRAppDistributionAuthPersistence clearAuthState:&error];
+  // TODO (schnecle): Add in FIRLogger to report when we have failed to clear auth state
+  if (error) {
+    NSLog(@"Error clearing token from keychain: %@", [error localizedDescription]);
+  }
   self.authState = nil;
   self.isTesterSignedIn = false;
 }
 
 - (void)fetchReleases:(FIRAppDistributionUpdateCheckCompletion)completion {
+  [self.authState performActionWithFreshTokens:^(NSString *_Nonnull accessToken,
+                                                 NSString *_Nonnull idToken,
+                                                 NSError *_Nullable error) {
+    if (error) {
+      // TODO (schnecle): Add in FIRLogger log statement
+      NSLog(@"Error fetching fresh tokens: %@", [error localizedDescription]);
+      [self signOutTester];
+      return;
+    }
 
-    [self.authState performActionWithFreshTokens:^(NSString *_Nonnull accessToken,
-                                               NSString *_Nonnull idToken,
-                                               NSError *_Nullable error) {
-      if (error) {
-        NSLog(@"Error fetching fresh tokens: %@", [error localizedDescription]);
-        [self signOutTester];
-        return;
-      }
+    // perform your API request using the tokens
+    NSURLSession *URLSession = [NSURLSession sharedSession];
+    NSMutableURLRequest *request = [[NSMutableURLRequest alloc] init];
+    NSString *URLString =
+        [NSString stringWithFormat:kReleasesEndpointURL, [[FIRApp defaultApp] options].googleAppID];
+    [request setURL:[NSURL URLWithString:URLString]];
+    [request setHTTPMethod:@"GET"];
+    [request setValue:[NSString stringWithFormat:@"Bearer %@", accessToken]
+        forHTTPHeaderField:@"Authorization"];
+
+    NSURLSessionDataTask *listReleasesDataTask = [URLSession
+        dataTaskWithRequest:request
+          completionHandler:^(NSData *data, NSURLResponse *response, NSError *error) {
+            if (error) {
+              // TODO: Reformat error into error code
+              completion(nil, error);
+              return;
+            }
+
+            NSHTTPURLResponse *HTTPResponse = (NSHTTPURLResponse *)response;
+
+            if (HTTPResponse.statusCode == 200) {
+              [self handleReleasesAPIResponseWithData:data completion:completion];
+            } else {
+              // TODO: Handle non-200 http response
+              NSLog(@"ERROR - Non 200 service response - %@", HTTPResponse);
+              @throw([NSException exceptionWithName:@"NotImplementedException"
```

Comment:
I'm assuming this is just a reminder, but would recommend to never ever throw in Objective-C

Reply:
Pranav is actively working on this 👍